docs/alerts.md

```diff
@@ -0,0 +1,312 @@
+This lends itself better to code generators because the `notify()` method can be customized using extern properties so that the act of checking and notifying can be separated from eachother.
+
+## Integration Methods
+
+The above constructs lend themselves well to various different methods of integration and also allow for building out future integration plans.
+
+### Inline
+
+It is possible for the entire query to run inline.
+
+```
+// Send to multiple notification handlers.
+notify = (tables=<-) => {
+	tables |> pagerduty.notify(routingKey: myRoutingKey)
+	tables |> slack.notify(channel: "#flux")
+}
+
+from(bucket: "telegraf")
+    |> range(start: -5m)
+    |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
+    |> check(
+    	crit: (r) => r._value > 90,
+    )
+    |> notify()
+```
+
+### Using Buckets
+
+Using the same script as the above, the `notify` method can be customized to write to a bucket instead and the notifications can be triggered off of the contents of the bucket instead.
+
+```
+// This goes in the check script.
+notify = (tables=<-) => tables |> to(bucket: "system")
+
+// This goes in a separate script.
+alert = (tables=<-, handler) => tables |> handler() |> to(bucket: "notifications")
+
+tables = from(bucket: "system")
+	|> range(start: -1m)
+	|> notify()
+	
+// The function is easy for the user to write and it is easy to
+// inject into a generated script.
+handler1 = (tables=<-) => pagerduty.notify(routingKey: myRoutingKey)
+handler2 = (tables=<-) => slack.notify(channel: "#flux")
+	
+// Send to multiple notification handlers.
+// This is easy to generate.
+tables |> handler1() |> to(bucket: "notifications")
+tables |> handler2() |> to(bucket: "notifications")
+```
+
+### Tasks
+
+Using the above, this can be integrated with tasks fairly easily and meet the above requirements.
+
+This might be the check script.
+
+```
+// Injected.
+option task = {
+	every: 1m,
+}
+
+// Possibly an option like now?
+notify = (tables=<-) => tables |> to(bucket: "system")
+
+// User provided.
+from(bucket: "telegraf")
+    |> range(start: -task.every)
+    |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system")
+    |> check(
+	crit: (r) => r._value > 90,
+    )
+    |> notify()
+```
+
+This might be the notification script.
+
+```
+// Injected.
+option task = {
+	every: 1m,
+}
+
+tables = from(bucket: "system")
+	|> range(start: -task.every)
+	|> notify()
+
+// The user could customize the function body for each handler,
+// but the function signature would probably be generated.
+handler1 = (tables=<-) => pagerduty.notify(routingKey: myRoutingKey)
+handler2 = (tables=<-) => slack.notify(channel: "#flux")
+
+// Send to multiple notification handlers.
+// This is easy to generate.
+tables |> handler1() |> to(bucket: "notifications")
+tables |> handler2() |> to(bucket: "notifications")
```

Comment:
Same as above, the API exposed by this package needs to write to the system bucket by default.

Reply:
I am confused. This is the notification log portion. The system bucket seems to only be involved for evaluating the checks. Did I misunderstand the spec?

Reply:
Both check statuses and notification logs need to be written to a system bucket. But which bucket is not too important for this design as we can easily make that parameterized. 

My initial comment was that this design requires that users explicitly write to a bucket, where we need that to be implicit (i.e. default) from the API.

Reply:
Ah. The idea here is that the user provided part goes into the implementation of `handler1` or `handler2`. With notifications, I am less clear on how that would be customized by power users. Do you have an example of how a user would want to do something more than just configure the handler using parameters?

It might make it easier for me to develop how this would work and write an example. Otherwise, it seems to me the handler portion is basically completely generated.